src/ig/IGExporter.ts

```diff
@@ -1338,96 +1342,202 @@ export class IGExporter {
+        parameter.code.system = 'http://hl7.org/fhir/tools/CodeSystem/ig-parameters';
+      }
+    });
+
+    // Add new copyrightLabel property
+    if (this.config.copyrightLabel) {
+      this.ig.copyrightLabel = this.config.copyrightLabel;
+    }
+
+    // Add new versionAlgorithm property
+    if (this.config.versionAlgorithmString) {
+      this.ig.versionAlgorithmString = this.config.versionAlgorithmString;
+    } else if (this.config.versionAlgorithmCoding) {
+      this.ig.versionAlgorithmCoding = this.config.versionAlgorithmCoding;
+    }
+
+    // Add new dependsOn.reason property
+    this.ig.dependsOn?.forEach(dependency => {
+      const configDependency = this.config.dependencies?.find(
+        d => d.packageId === dependency.packageId
+      );
+      if (configDependency.reason) {
+        dependency.reason = configDependency.reason;
+      }
+    });
+  }
+
+  // If an R4 IG uses any new properties from R5, they should be included as extensions.
+  // Converting between FHIR versions is documented here: http://hl7.org/fhir/2022Sep/versions.html#extensions
+  // If an R4 IG uses any R5 properties that exist in R4 but is different in R5, the R4 property is used.
+  translateR5PropertiesToR4() {
+    this.ig.definition.resource.forEach(resource => {
+      // Assign IG.definition.resource.profile to exampleCanonical if it is not set, otherwise add an extension
+      const configEntry = this.config.resources?.find(
+        r => r.reference?.reference === resource.reference?.reference
+      );
+      if (resource.exampleCanonical == null) {
         if (configEntry?.profile != null) {
-          resource.profile = configEntry.profile;
+          if (configEntry.profile.length === 1) {
+            resource.exampleCanonical = configEntry.profile[0];
+            delete resource.exampleBoolean;
+          } else if (configEntry.profile.length > 1) {
+            resource.exampleCanonical = configEntry.profile[0];
+            delete resource.exampleBoolean;
+            resource.extension = (resource.extension ?? []).concat({
+              url: 'http://hl7.org/fhir/5.0/StructureDefinition/extension-ImplementationGuide.definition.resource.profile',
+              valueCanonical: configEntry.profile
+            });
+          }
         }
-        // Assign Ig.definition.resource.isExample if provided
-        if (configEntry?.isExample != null) {
-          resource.isExample = configEntry.isExample;
+      } else {
+        if (
+          configEntry?.profile != null &&
+          configEntry.profile.length === 1 &&
+          configEntry.profile[0] !== resource.exampleCanonical
+        ) {
+          resource.extension = (resource.extension ?? []).concat({
+            url: 'http://hl7.org/fhir/5.0/StructureDefinition/extension-ImplementationGuide.definition.resource.profile',
+            valueCanonical: configEntry.profile
+          });
         }
-      });
+      }
```

Comment:
I think lines 1445 - 1470 above can be done more succinctly like so:
```ts
      if (resource.exampleCanonical == null && configEntry?.profile?.length) {
        resource.exampleCanonical = configEntry.profile[0];
        delete resource.exampleBoolean;
      }
      if (
        resource.exampleCanonical != null &&
        configEntry?.profile?.some(p => p !== resource.exampleCanonical)
      ) {
        resource.extension = (resource.extension ?? []).concat({
          url: 'http://hl7.org/fhir/5.0/StructureDefinition/extension-ImplementationGuide.definition.resource.profile',
          valueCanonical: configEntry.profile
        });
      }
```

I tried it locally and all the tests still pass.